apps/ledger-live-mobile/src/screens/Swap/Form/Max.tsx

```diff
@@ -1,14 +1,27 @@
-import React from "react";
+import React, { useEffect, useMemo } from "react";
 import { useTranslation } from "react-i18next";
 import { Flex, Text, Switch } from "@ledgerhq/native-ui";
 import { SwapTransactionType } from "@ledgerhq/live-common/exchange/swap/types";
+import { isAccount } from "@ledgerhq/live-common/account/index";
 import { useAnalytics } from "~/analytics";
 import { sharedSwapTracking } from "../utils";
 
 export function Max({ swapTx }: { swapTx: SwapTransactionType }) {
   const { t } = useTranslation();
   const { track } = useAnalytics();
 
+  const isMaxButtonDisabled = useMemo(
+    () => isAccount(swapTx.swap.from.account),
+    [swapTx.swap.from.account],
+  );
+
+  useEffect(() => {
+    if (isMaxButtonDisabled && swapTx.swap.isMaxEnabled) {
+      swapTx.toggleMax();
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [isMaxButtonDisabled, swapTx.swap.isMaxEnabled, swapTx.swap.from.amount]);
```

Comment:
i'm not familiar with what this code is doing, but as far as I understand it you try to keep the max toggled off and synchronized based on the value of isMaxButtonDisabled ?
what would be the reason to have the from.amount in the deps, as it's not used? ultimately is there a possibility that `from.amount` changes twice in a raw and trigger a double toggleMax?

because on one side you have booleans (isMaxButtonDisabled, isMaxEnabled) but on the other side you have a function that toggles something based on them (which itself is also asynchronous, looking at the code of `toggleMax` in libs/ledger-live-common/src/exchange/swap/hooks/useUpdateMaxAmount.ts)

thanks